src/firmware/app/control/trajectory_planner.h

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+#include "firmware/shared/math/polynomial_2d.h"
+#include "firmware/shared/math/vector_2d.h"
+
+#define TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS                                              \
+    6000  // The maximum size of the array containing trajectory elements
+#define __TRAJECTORY_PLANNER_MAX_PATH_LENGTH__ 18  // [meters]
+
+// Struct that defines a single point on a trajectory
+// Includes the Position,and Timedata corresponding to that point
+typedef struct TrajectoryElement
+{
+    Vector2d_t position;
+    double time;
```

Comment:
We can't use `double`'s on the arm processor because it's only 32 bit. I think the compiler does something funky with rounding to make it work, but let's just be explicit.

Reply:
Switched to Floats

Reply:
We should change all the `EXPECT_DOUBLE_EQ` to `EXPECT_FLOAT_EQ` as well eh? (pretty sure that's a gtest function).

Reply:
Changed to EXPECT_FLOAT_EQ